### PR TITLE
[ROCm] enable missing dtypes in unit tests

### PR DESCRIPTION
**Pull request by arindamroy-eng**
Added more data types for ROCm by updating dtypesIfROCM to match dtypesIfCUDA.
Additionally added bfloat16 to the commit.

### CODE REVIEW

**Comment by arindamroy-eng** [comment]
@jeffdaily

File: torch/testing/_internal/common_methods_invocations.py [hunk 1]
```diff
@@ -1705,7 +1705,8 @@ def _make_tensor_helper(shape, low=None, high=None):
            # BFloat16 support on CUDA requires CUDA 11 and SM53
            dtypesIfCUDA=floating_types_and(torch.float16, torch.complex64, torch.complex128,
                                            *[torch.bfloat16] if CUDA11OrLater else []),
-           dtypesIfROCM=floating_types_and(torch.half),
+           dtypesIfROCM=floating_types_and(torch.float16, torch.complex64, torch.complex128,
+                                           *[torch.bfloat16] if CUDA11OrLater else []),
```
Comment (jeffdaily) [inline torch/testing/_internal/common_methods_invocations.py:1709]:
```suggestion
           dtypesIfROCM=floating_types_and(torch.float16, torch.complex64, torch.complex128),
```
Let's leave out bfloat16 for now.  The `CUDA11OrLater` condition will always be false for HIP; removing it makes it more clear.

Comment (jeffdaily) [reply torch/testing/_internal/common_methods_invocations.py:1709]: New suggestion coming.

Comment (jeffdaily) [inline torch/testing/_internal/common_methods_invocations.py:1709]:
```suggestion
           dtypesIfROCM=floating_types_and(torch.float16, torch.complex64, torch.complex128, torch.bfloat16),
```

Comment (jeffdaily) [reply torch/testing/_internal/common_methods_invocations.py:1709]: We support bfloat16 today, so we should be testing for it.

**Review by jeffdaily — APPROVED** [review]
LGTM so long as CI passes

**Comment by arindamroy-eng** [comment]
Will be submitting a new PR with the issues resolved.

**Pull request closed without merging**